src/gui/dialogs/end_credits.cpp

```diff
@@ -73,11 +63,12 @@ void end_credits::pre_show(window& window)
 {
 	window.set_callback_next_draw([this]()
 	{
-		timer_id_ = add_timer(10, std::bind(&end_credits::timer_callback, this), true);
 		// Delay a little before beginning the scrolling
 		last_scroll_ = SDL_GetTicks() + 3000;
 	});
 
+	window.connect_signal<event::DRAW>(std::bind(&end_credits::timer_callback, this), event::dispatcher::front_child);
```

Comment:
You just added an uninitialized memory read.

The window's callback_next_draw is called **after** drawing the window, and therefore after event::DRAW has been sent. Thus, at the time of the first frame, `last_scroll_` isn't set. Because `timer_callback()` reads `last_scroll_`, it triggers undefined behavior.

Please initialize `last_scroll_` in the constructor initializer list. I suggest setting it to the highest possible uint32_t value because we don't want the first frame to scroll:

    , last_scroll_(std::numeric_limits<uint32_t>::max());